internal/codeintel/dependencies/internal/lockfiles/dependency_graph.go

```diff
@@ -59,32 +58,89 @@ func (dg *DependencyGraph) AllEdges() (edges []Edge) {
 	return edges
 }
 
+type pkgSet = map[reposource.PackageVersion]struct{}
+
 func (dg *DependencyGraph) String() string {
 	var out strings.Builder
 
+	json.NewEncoder(&out).Encode(dg.AsMap())
+
+	return out.String()
+}
+
+func (dg *DependencyGraph) AsMap() map[string]interface{} {
```

Comment:
```suggestion
func (dg *DependencyGraph) AsMap() map[string]any {
```
just a minor thing :p